examples/helloworld/lib/server.ex

```diff
@@ -1,5 +1,43 @@
+defmodule FooReplacer do
+  def call(service_mod, stream, {_, {req_mod, req_stream}, {res_mod, res_stream}} = rpc, func_name, next) do
```

Comment:
I just think there's many arguments for a middleware. Is there anyway to make it simple?

Reply:
Yeah I do agree. I just wanted to keep things simple for a start but it's a bit clunky as it is and requires to dive in the internals to understand what's going on. 

We could [like Go does](https://github.com/grpc/grpc-go/blob/d1cd1551aa4cd02197f5715f82f9a4df8e053bc7/interceptor.go#L26), just take move the middleware further down the line, after the request / response had been deserialized. 

But that's a decision I'd rather talk with you first 😄

Reply:
In theory, we can remove `stream`(if we don't want to support overriding marshal/unmarshal) and `func_name`. `server` in stream is one-to-one with server, it's enough for users to log something with the service. `adapter` and `payload` is specified to the adapter, most users probably not want to access to those. And `func_name` can be calculated using name in rpc tuple.